src/framework/bot/listener.ts

```diff
@@ -106,6 +109,55 @@ export class Listener {
         client.on('warn', this.onWarn.bind(this));
     }
 
+    /**
+     * Stops the listeners.
+     */
+    public stop() {
+        let { Framework } = require('../framework');
+        let client = Framework.client as Client;
+
+        // First the static ones
+        client.removeListener('channelCreate', this.onChannelCreate.bind(this));
+        client.removeListener('channelDelete', this.onChannelDelete.bind(this));
+        client.removeListener('channelPinsUpdate', this.onChannelPinsUpdate.bind(this));
+        client.removeListener('channelUpdate', this.onChannelUpdate.bind(this));
+        client.removeListener('debug', this.onDebug.bind(this));
+        client.removeListener('disconnect', this.onDisconnect.bind(this));
+        client.removeListener('emojiCreate', this.onEmojiCreate.bind(this));
+        client.removeListener('emojiDelete', this.onEmojiDelete.bind(this));
+        client.removeListener('emojiUpdate', this.onEmojiUpdate.bind(this));
+        client.removeListener('error', this.onError.bind(this));
+        client.removeListener('guildBanAdd', this.onGuildBanAdd.bind(this));
+        client.removeListener('guildBanRemove', this.onGuildBanRemove.bind(this));
+        client.removeListener('guildMembersChunk', this.onGuildMembersChunk.bind(this));
+        client.removeListener('guildMemberSpeaking', this.onGuildMemberSpeaking.bind(this));
+        client.removeListener('guildUnavailable', this.onGuildUnavailable.bind(this));
+        client.removeListener('guildUpdate', this.onGuildUpdate.bind(this));
+        client.removeListener('messageDeleteBulk', this.onMessageDeleteBulk.bind(this));
+        client.removeListener('messageReactionAdd', this.onMessageReactionAdd.bind(this));
+        client.removeListener('messageReactionRemove', this.onMessageReactionRemove.bind(this));
+        client.removeListener('messageReactionRemoveAll', this.onMessageReactionRemoveAll.bind(this));
+        client.removeListener('messageUpdate', this.onMessageUpdate.bind(this));
+        client.removeListener('presenceUpdate', this.onPresenceUpdate.bind(this));
+        client.removeListener('rateLimit', this.onRateLimit.bind(this));
+        client.removeListener('reconnecting', this.onReconnecting.bind(this));
+        client.removeListener('resume', this.onResume.bind(this));
+        client.removeListener('roleCreate', this.onRoleCreate.bind(this));
+        client.removeListener('roleDelete', this.onRoleDelete.bind(this));
+        client.removeListener('roleUpdate', this.onRoleUpdate.bind(this));
+        client.removeListener('userUpdate', this.onUserUpdate.bind(this));
+        client.removeListener('voiceStateUpdate', this.onVoiceStateUpdate.bind(this));
+        client.removeListener('warn', this.onWarn.bind(this));
```

Comment:
Maybe there's a better way to write this out than writing client.removeListener on every line of this. I don't see a better alternative to client.removeListener coming up anytime soon, so it probably doesn't matter anyways.